### PR TITLE
fix: Fix DeadStoreProcessor deleting invocations

### LINKED ISSUE #412: DeadStoreProcessor sometimes processes wrong element
I knew we'd run into some kind of problem entirely relying on the new position matching.

The DeadStoreProcessor handles elements of type CtStatement. That's too broad. For example, `CtInvocation` is a `CtStatement`, so if this is a dead store:

```java
log = tempLog.newChildLog();
```
Even though the invocation isn't in a statement position here, the processor might delete only the invocation, leaving the syntactically invalid:

```java
log = ;
```

Here, we need to steer the matching with `canRepairInternal` to _only_ consider `CtLocalVariable` and `CtAssignment`. Those are the only two kinds of elements that can cause dead stores. Technically, a `CtField` can also have a dead store, but Sonar does not appear to care about that.

### PR DESCRIPTION
**Pull request by slarse**
Fix #412 

See the issue for a description of the problem. The fix is trivial, see e1c91ca.

### CODE REVIEW

File: src/main/java/sorald/processor/DeadStoreProcessor.java [hunk 1]
```diff
@@ -20,6 +20,11 @@
 @ProcessorAnnotation(key = 1854, description = "Unused assignments should be removed")
 public class DeadStoreProcessor extends SoraldAbstractProcessor<CtStatement> {
 
+    @Override
+    protected boolean canRepairInternal(CtStatement candidate) {
+        return candidate instanceof CtLocalVariable || candidate instanceof CtAssignment;
```
[the remark this answers is not in the document]
Comment (slarse) [reply src/main/java/sorald/processor/DeadStoreProcessor.java:25]: All noncompliant lines in this test file are assignments:

https://github.com/SpoonLabs/sorald/blob/master/src/test/resources/processor_test_files/1854_DeadStore/DeadStores.java

File: src/test/resources/processor_test_files/1854_DeadStore/DeadStoreWithInvocation.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
+As a CtInvocation is a CtStatement, and the DeadStoreProcessor processes CtStatement,
+it is important to test cases of dead stores with CtInvocation's as the assigned expr.
+See #412 for a related bug.
+ */
+
+public class DeadStoreWithInvocation {
+    public int deadInitializer() {
+        int a = Integer.parseInt("1"); // Noncompliant
```
[the remark this answers is not in the document]
Comment (slarse) [reply src/test/resources/processor_test_files/1854_DeadStore/DeadStoreWithInvocation.java:9]: No, this is a `CtLocalVariable`

File: src/test/resources/processor_test_files/1854_DeadStore/DeadStoreWithInvocation.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        a = 2;
+        return a;
+    }
+
+    public void deadAssignment() {
+        int a = 2;
+        System.out.println(a);
+        a = Integer.parseInt("1"); // Noncompliant
```
Comment (slarse) [inline src/test/resources/processor_test_files/1854_DeadStore/DeadStoreWithInvocation.java:17]: This is a CtAssignment

File: src/main/java/sorald/processor/DeadStoreProcessor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (slarse) [reply src/main/java/sorald/processor/DeadStoreProcessor.java:25]: Also, see my comment below.

[the remark this answers is not in the document]
Comment (khaes-kth) [reply src/main/java/sorald/processor/DeadStoreProcessor.java:25]: Hmm.
Looking at your example in #412 , I thought we only pass the right hand side of the assignment to this processor (which would not be wise!)
Now I see what you are doing.

[the remark this answers is not in the document]
Comment (slarse) [reply src/main/java/sorald/processor/DeadStoreProcessor.java:25]: The problem exemplified in #412 is that the assigned expression itself is an invocation in expression position. Invocations are however _also_ statements, and are thus amenable for processing by the processor. This is what caused the bug.

[the remark this answers is not in the document]
Comment (khaes-kth) [reply src/main/java/sorald/processor/DeadStoreProcessor.java:25]: Yeah, I got it.

[the remark this answers is not in the document]
Comment (slarse) [reply src/main/java/sorald/processor/DeadStoreProcessor.java:25]: :+1:

**Pull request merged**